dataland-api-key-manager/src/main/kotlin/org/dataland/datalandapikeymanager/services/ApiKeyManager.kt

```diff
@@ -46,19 +51,31 @@ class ApiKeyManager(
     }
 
     private fun checkIfDaysValidValueIsValid(daysValid: Int?) {
-        if (daysValid != null && daysValid <= 0) {
-            throw InvalidInputApiException(
-                "If set, the value of daysValid must be a positive integer.",
-                "If set, the value of daysValid must be a positive integer but it was $daysValid",
-            )
+        if (daysValid != null) {
+            if (daysValid < 1) {
+                throw InvalidInputApiException(
+                    "If set, the value of daysValid must be a positive number.",
+                    "If set, the value of daysValid must be a positive number but it was $daysValid",
+                )
+            }
+            if (daysValid > maxDaysOfValidityForApiKey) {
+                throw InvalidInputApiException(
+                    "If set, the value of daysValid cannot be greater than 3650.",
+                    "If set, the value of daysValid cannot be greater than 3650 but it was $daysValid",
+                )
+            }
         }
     }
 
     private fun calculateExpiryDate(daysValid: Int?): Long? {
         checkIfDaysValidValueIsValid(daysValid)
         return when (daysValid) {
             null -> null
-            else -> (daysValid * milliSecondsInADay) + Instant.now().toEpochMilli()
+            else -> {
+                val daysValidLong: Long = daysValid.toLong()
```

Comment:
Ich denke nicht, dass es die `*Long` Suffizes brauch.

Reply:
`Instant.now().toEpochMilli() + (daysValid * milliSecondsInADay)` would also work but yours is more readable

Reply:
Suffix kann ich rausnehmen.

`Instant.now().toEpochMilli() + (daysValid * milliSecondsInADay)` klappt tatsächlich nicht. Genau das war ja der Bug.
Er castet nämlich das Produkt in der Klammer als integer, und das führt zum Overflow.

Ich könnte:
`Instant.now().toEpochMilli() + (daysValid.toLong() * milliSecondsInADay)` 
machen. Da hatte ich aber eine Mini Angst im Hinterkopf, dass er das Produkt als integer interpretieren könnte, weil `milliSecondsInADay` ein `Int` ist.

Reply:
( Wegen der Angst im Hinterkopf habe ich hier ganz fein säuberlich und explizit erst zwei Longs erstellt, und die erst dann multipliziert)

Reply:
Absolut berechtigt. Ich hätte aber gedacht, dass `Instant.now().toEpochMilli() + (daysValid * milliSecondsInADay)`, weil der Long als erstes kommt und darauf addiert wird. Ursprünglich war es ja vertauscht.

Reply:
Komisch. Im Kotlin Playground sind beide Varianten ein Long 😮.

Reply:
Hättest du jetzt gerne eine Änderung hier?  Oder soll ich die sehr explizite Schreibweise so lassen wie sie ist?

Reply:
Lol, ich meine eigentlich eher `daysValidLong` => `daysValid`. Es gibt Leute, die der Meinung sind, dass das a) redundant ist und b) den Leseflow stört.

Reply:
Will do